amy/extrequests/views.py

```diff
@@ -571,53 +601,49 @@ def all_trainingrequests(request):
+                            membership=membership_to_use,
+                            seat_public=seat_public,
+                            event=event,
+                        )
+                    ]
+
+            # if membership is the same for all matches,
+            # collect warnings after all requests are processed
             if membership:
-                remaining = (
-                    membership.public_instructor_training_seats_remaining
-                    if seat_public
-                    else membership.inhouse_instructor_training_seats_remaining
+                warnings = get_membership_warnings_after_match(
+                    membership=membership, seat_public=seat_public, event=event
                 )
-                if remaining <= 0:
-                    messages.warning(
-                        request,
-                        f'Membership "{membership}" is using more training seats than '
-                        "it's been allowed.",
-                    )
 
-                # check if membership is active
-                if not (
-                    membership.agreement_start <= today <= membership.agreement_end
-                ):
-                    messages.warning(
-                        request,
-                        f'Membership "{membership}" is not active.',
-                    )
-
-                # show warning if training falls out of agreement dates
-                if (
-                    event.start
-                    and event.start < membership.agreement_start
-                    or event.end
-                    and event.end > membership.agreement_end
-                ):
-                    messages.warning(
-                        request,
-                        f'Training "{event}" has start or end date outside '
-                        f'membership "{membership}" agreement dates.',
-                    )
-
-            messages.success(
-                request,
-                "Successfully accepted and matched selected people to training.",
-            )
+            for msg in warnings:
+                messages.warning(request, msg)
+            if errors:
+                for msg in errors:
+                    messages.error(request, msg)
+                changed_count = len(match_form.cleaned_data["requests"]) - len(errors)
+                info_msg = (
+                    f"Accepted and matched {changed_count} "
+                    f'{"person" if changed_count==1 else "people"} to training, '
+                    f"which raised {len(warnings)} warning(s). "
+                    f"{len(errors)} request(s) were skipped due to errors."
+                )
+                messages.info(request, info_msg)
+            else:
+                messages.success(
+                    request,
+                    "Successfully accepted and matched selected people to training.",
+                )
```

Comment:
I don't like how how view is expanding, but to be honest I spent like 30 minutes trying to restructure this code and the results were not satisfactory.

If you have some ideas how we could improve this logic, let me know.

Things I like:
1. separate function to determine warnings, `get_membership_warnings_after_match`

Things concerning me:
1. use of `if membership_auto_assign` twice, then use of `if membership` (which is basically the same as `if not membership_auto_assign`) - this seems like two different logic flows intertwined
2. errors not calculated in separate function (just like warnings)
3. not a lot of code moved outside to separate functions (makes testing difficult)